beater/pubsubbeat.go

```diff
@@ -100,6 +105,46 @@ func (bt *Pubsubbeat) Run(b *beat.Beat) error {
 		bt.pubsubClient.Close()
 	}()
 
+	events := make(chan ackableEvent)
+
+	go func() {
+		// batch up documents to prevent contending on lock in client.Publish()
+
+		batchMaxSize := 512
+		batchMaxTime := 1 * time.Second
+
+		t := time.Tick(batchMaxTime)
```

Comment:
I used a ticker here instead of a `time.After()` in the loop in order to avoid allocating that channel in the critical path. That does mean that technically it will fire every second, that doesn't seem too problematic though. Happy to change it though!